Mathlib/RingTheory/DedekindDomain/Factorization.lean

```diff
@@ -28,11 +51,32 @@ open scoped BigOperators Classical nonZeroDivisors
 
 open Set Function UniqueFactorizationMonoid IsDedekindDomain IsDedekindDomain.HeightOneSpectrum
 
-/-! ### Factorization of ideals of Dedekind domains -/
+variable {R : Type*} [CommRing R] {K : Type*} [Field K] [Algebra R K] [IsFractionRing R K]
+
+namespace FractionalIdeal
+
+/-- If `I` is a nonzero fractional ideal, `a ∈ R`, and `J` is an ideal of `R` such that
+`I = a⁻¹J`, then `J` is nonzero. -/
+theorem ideal_factor_ne_zero {I : FractionalIdeal (nonZeroDivisors R) K} (hI : I ≠ 0) {a : R}
```

Comment:
There is a notation `R⁰` for `nonZeroDivisors R` if you `open nonZeroDivisors`. The code would be easier to read if you use it.